src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilTest.java

```diff
@@ -166,7 +166,18 @@ public void testIsLocalVariableDefUnexpected() {
 
     @Test
     public void testIsLocalVariableDefResource() {
-        assertTrue(ScopeUtil.isLocalVariableDef(getNode(TokenTypes.RESOURCE)), "invalid result");
+        final DetailAstImpl resourceTwoChildren = getNode(TokenTypes.RESOURCE);
+        final DetailAstImpl modifiers = new DetailAstImpl();
+        modifiers.setType(TokenTypes.MODIFIERS);
```

Comment:
Can we avoid this type of tests ? 
We did do much to avoid this level of fakes that make no good to code.

Reply:
This method checks node, so we need to provide a node to test. Another way is to parse some file, navigate to node and check result.
I do not like the way test is written, but it is common approach for this util class.
I suggest to create separate issue for refactoring of this test class to avoid using fake nodes.

Reply:
what force you to create this test ?
We had several bad cases of test to satisfy pitest coverage that it split in several groups, so some cases claimed to be not reachable.

Reply:
I am not the one who created this test, it was like this before. I modified the logic, and result for such dummy node (with no children) is changed. This will never happen in real life. Other conditions (1 child, 2+ children) are covered by test cases for checks, which is not good since it is indirect test.
So I added new cases to check changed logic directly.

Reply:
one more note - there is no need for this test case. Pitest will pass because this condition is tested indirectly by test cases for several checks. 
@romani if this is fine, I can remove it. If not, such test case should be kept to test this util method directly.

Reply:
Please remove this test, we need to keep pure UT only as very very last resort to satisfy coverage